process_pr.py

```diff
@@ -230,31 +229,63 @@ def collect_commit_cache(bot_cache):
         bot_cache["commits"] = commit_cache
 
 
-def read_bot_cache(comment_msg):
+def read_bot_cache(data):
+    print("Loading bot cache")
+    res = loads_maybe_decompress(data)
+    for k, v in BOT_CACHE_TEMPLATE.items():
+        if k not in res:
+            res[k] = v
+    collect_commit_cache(res)
+    return res
+
+
+def extact_bot_cache(comment_msg):
     seen_commits_match = REGEX_COMMITS_CACHE.search(comment_msg)
     if seen_commits_match:
-        print("Loading bot cache")
-        res = loads_maybe_decompress(seen_commits_match[1])
-        for k, v in BOT_CACHE_TEMPLATE.items():
-            if k not in res:
-                res[k] = v
-        collect_commit_cache(res)
-        return res
+        return read_bot_cache(seen_commits_match[1])
     return {}
 
 
-def write_bot_cache(bot_cache, cache_comment, issue, dryRun):
-    old_body = cache_comment.body if cache_comment else CMSBOT_TECHNICAL_MSG
-    new_body = (
-        REGEX_COMMITS_CACHE.sub("", old_body)
-        + "<!-- bot cache: "
-        + dumps_maybe_compress(bot_cache)
-        + " -->"
-    )
-    if old_body == new_body:
-        return
-    print("Saving bot cache")
-    if len(new_body) <= 65535:
+def extract_bot_cache_v2(comment_msgs):
+    data = ""
+    for comment_msg in comment_msgs:
+        seen_commits_match = REGEX_COMMITS_CACHE.search(comment_msg.body)
+        if seen_commits_match:
+            data += seen_commits_match[1]
+
+    if data:
+        return read_bot_cache(data)
+
+    return {}
+
+
+def prepare_bot_cache(bot_cache):
+    res = []
+    data = dumps_maybe_compress(bot_cache)
+    while data:
+        # The limit is 65535 chars, and minimal bot cache comment is"cms-bot internal usage0<!-- bot cache:  -->",
+        # 43 characters. This leaves us 65492 chars for actual data. Keeping 2 extra characters in case bot cache
+        # spans more than 10 commits (usage0...usage9)
+        part, data = data[:55000], data[55000:]
+        res.append(part)
+
+    return res
+
+
+def write_bot_cache(bot_cache, cache_comments, issue, dryRun):
+    data = prepare_bot_cache(bot_cache)
+    for i, part in enumerate(data):
+        try:
+            cache_comment = cache_comments[i]
+        except (TypeError, IndexError):
+            cache_comment = None
+
+        old_body = cache_comment.body if cache_comment else CMSBOT_TECHNICAL_MSG
+        new_body = REGEX_COMMITS_CACHE.sub("", old_body) + "<!-- bot cache: " + part + " -->"
+        if old_body == new_body:
+            return
```

Comment:
@iarspider , this should be continue instead of return .... right?

Reply:
I think it doesn't matter, unless we get extremely (un)lucky. The only case where it matters is if the cache is big enough to exceed one comment and one of the chunks of base64-encoded compressed cache is exactly the same as before, but overall base64-encoded compressed cache is not.

Reply:
if it does not matter then I would prefer to process all parts to make sure the contents are up to date

Reply:
Changed just in case.